aioamqp/channel.py

```diff
@@ -109,7 +123,12 @@ def open(self, timeout=None):
         request = amqp_frame.AmqpEncoder()
         request.write_shortstr('')
         fut = self._set_waiter('open')
-        yield from self._write_frame(frame, request, no_wait=False, timeout=timeout, no_check_open=True)
```

Comment:
Definitely. I was thinking about it, but didn't know what coding style you prefer. Maybe the _write_frame method should take a string argument and call self._set_waiter with it, returning the future and handing the exception. The waiter is created on the ~ preceding line anyway.